### PR TITLE
Tensorflow QA example

### PR DESCRIPTION
**Pull request by Rocketknight1**
The new TF QA example! There were a couple of issues with the metrics that might need more investigation, but I confirmed they happened in the PyTorch version too. Possibly that was caused by evaluating on an untrained model, though.

Also, don't stress about reviewing this until after the weekend!

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Great job wrapping this up! I suspect there are some references to `run_tf_squad.py` in the docs - could you verify if that's so, and if it is update the references to point to this version?

File: examples/tensorflow/question-answering/README.md [hunk 1]
```diff
@@ -14,21 +14,43 @@ See the License for the specific language governing permissions and
...
+
+This folder contains the `run_qa.py` script, demonstrating *question answering* with the 🤗 Transformers library.
+For straightforward use-cases you may be able to use this script without modification, although we have also
+included comments in the code to indicate areas that you may need to adapt to your own projects. 
+
+### Usage notes
+Note that when contexts are long they may be split into multiple training cases, not all of which may contain
+the answer span. 
+
+As-is, the example script will train on SQuAD or any other question-answering dataset, and can handle user
```
Comment (sgugger) [inline examples/tensorflow/question-answering/README.md:27]:
```suggestion
As-is, the example script will train on SQuAD or any other question-answering dataset formatted the same way, and can handle user
```

File: examples/tensorflow/question-answering/utils_qa.py [hunk 2]
```diff
@@ -0,0 +1,425 @@
+# coding=utf-8
```
Comment (sgugger) [inline examples/tensorflow/question-answering/utils_qa.py:1]: Is this a direct copy of the PyTorch utils_qa.py? If yes, we should find a way to have both stay consistent with each other.

**Review by sgugger — APPROVED** [review]
Great to have this new example!

File: examples/tensorflow/question-answering/utils_qa.py (hunk 2 above)
Comment (Rocketknight1) [reply examples/tensorflow/question-answering/utils_qa.py:1]: It is! I know we talked about this before, but what's the right way to do that? I don't think we can symlink the same file in two places with git repos, but do we have some kind of macros we use to keep them consistent?

Comment (sgugger) [reply examples/tensorflow/question-answering/utils_qa.py:1]: We have utilities that check copies internally, will add this to it when I have a bit of time.

Comment (Rocketknight1) [reply examples/tensorflow/question-answering/utils_qa.py:1]: Cool, thanks! I'll try to learn the syntax from how you do it and do it myself in future.

**Pull request merged**